okhttp/src/main/kotlin/okhttp3/internal/cache/DiskLruCache.kt

```diff
@@ -697,7 +726,11 @@ class DiskLruCache(
   @Throws(IOException::class)
   fun delete() {
     close()
-    fileSystem.deleteContents(directory)
+    if (fileSystem.exists(directory)) {
+      fileSystem.deleteRecursively(directory)
+      // TODO check this logic
+      fileSystem.createDirectory(directory)
```

Comment:
This isn’t right because the permissions and other directory metadata might change. Instead you gotta loop over `directory` children and delete each